PSix/FBRequest.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FBRequest.swift
+//  PSix
+//
+//  Created by Turzion, Avihu on 4/20/15.
+//  Copyright (c) 2015 PayPal. All rights reserved.
+//
+
+import Foundation
+import Parse
+
+
+typealias FBParamsList = [NSObject: AnyObject]
+
+let URI_SEP = "/"
+
+
+class FBRequest {
+    
+    private(set) var path: String
+    private(set) internal var params: FBParamsList
+    
+    init(path: String, params: FBParamsList = FBParamsList()) {
+        self.path = path
+        self.params = params
+    }
+    
+    func addParam(param: String, value: AnyObject) -> FBRequest {
+        params[param] = value
+        return self
+    }
+    
+    func edge(name: String) -> FBRequest {
+        return FBRequest(path: path + URI_SEP + name)
```

Comment:
In a way that's the desired behavior. Because the parameters change depending on the edge you're referencing. The better thing to do would be to error when an edge is referenced after parameters were specified.